docker/build-tools/Dockerfile

```diff
@@ -101,59 +133,32 @@ RUN GO111MODULE=on go get istio.io/tools/protoc-gen-docs
 RUN GO111MODULE=on go get istio.io/tools/cmd/annotations_prep
 RUN GO111MODULE=on go get istio.io/tools/openapi/cue
 
-# Put the stuff we need in its final output location
+# Compress the Go tools and put them in their final location
+ADD https://github.com/upx/upx/releases/download/v${UPX_VERSION}/upx-${UPX_VERSION}-amd64_linux.tar.xz /tmp
+RUN tar -xJf upx-${UPX_VERSION}-amd64_linux.tar.xz -C /tmp
+RUN mv /tmp/upx-${UPX_VERSION}-amd64_linux/upx /usr/bin
+RUN upx --lzma /go/bin/*
```

Comment:
this isn't compressing all of the tools.  I'd instead compress `${OUTDIR}/usr/bin` (this misses protoc) but can be handled in a followon.